silq/parameters/acquisition_parameters.py

```diff
@@ -1928,10 +1934,19 @@ def analyse(self, traces: Dict[str, Dict[str, np.ndarray]] = None):
                 up_proportions[f_idx, sample] = read_result['up_proportion']
                 results['results_read'].append(read_result)
 
+            state_probability_0[f_idx] = np.sum(up_proportions[f_idx] <
+                                                self.threshold_up_proportion)/self.samples
+            state_probability_1[f_idx] = np.sum(up_proportions[f_idx] >
+                                                self.threshold_up_proportion)/self.samples
+
             if len(self.ESR_frequencies) > 1:
                 results[f'up_proportions_{f_idx}'] = up_proportions[f_idx]
+                results[f'state_probability_0_{f_idx}'] = state_probability_0[f_idx]
+                results[f'state_probability_1_{f_idx}'] = state_probability_1[f_idx]
             else:
                 results['up_proportions'] = up_proportions[f_idx]
+                results['state_probability_0'] = state_probability_0[f_idx]
+                results['state_probability_1'] = state_probability_1[f_idx]
```

Comment:
Here I think we can just store `state_probability` or `success_probability` directly to the dataset without splitting it out based on `f_idx`, it will just be a 2D array. I believe Serwan has already implemented something like this for the new measurement parameters.

We can check in on that later.

Reply:
But isn't it worth having this similar to flip_probability? I.e. when we readout separately different ESR peaks, we calculate flip probability and store this separately. The same I would guess to be for this "success_probability"